app/bundles/CoreBundle/Service/GlobalSearch.php

```diff
@@ -0,0 +1,135 @@
+                'expr'   => 'eq',
+                'value'  => $this->userHelper->getUser()->getId(),
+            ];
+        }
+
+        $results = $model->getEntities([
+            'filter'           => $filter,
+            'start'            => 0,
+            'limit'            => GlobalSearchEvent::RESULTS_LIMIT,
+            'ignore_paginator' => true,
+            'with_total_count' => true,
+        ]);
+
+        return $this->processResults($results, $searchString, $template, $permissions);
+    }
+
+    /**
+     * Process search results and render templates.
+     *
+     * @param array<mixed>         $results
+     * @param array<string, bool> $permissions
+     *
+     * @return array<int|string, mixed>
+     */
+    private function processResults(
+        array $results,
+        string $searchString,
+        string $template,
+        array $permissions,
+    ): array {
+        $count = $results['count'] ? (int) $results['count'] : 0;
+
+        if ($count === 0) {
+            return [];
+        }
+
+        $renderedResults = array_map(
+            fn($item) => $this->twig->render($template, [
+                'item'    => $item,
+                'canEdit' => $this->canEditItem($item, $permissions),
+            ]),
+            $results['results']
+        );
+
+        if ($count > GlobalSearchEvent::RESULTS_LIMIT) {
+            $renderedResults[] = $this->twig->render($template, [
+                'searchString' => $searchString,
+                'showMore'     => true,
+                'remaining'    => $count - GlobalSearchEvent::RESULTS_LIMIT,
+            ]);
+        }
+
+        $renderedResults['count'] = $count;
+
+        return $renderedResults;
+    }
+
+    /**
+     * Determine if the current user can edit an item.
+     *
+     * @param object              $item
+     * @param array<string, bool> $permissions
+     *
+     * @return bool
+     */
+    private function canEditItem(object $item, array $permissions): bool
+    {
+        return $item->getCreatedBy() === $this->userHelper->getUser()->getId()
+            ? $permissions['editown']
+            : $permissions['editother'];
+    }
+
+    /**
+     * Retrieve permissions based on the given base and skip flag.
+     *
+     * @param string $permissionsBase
+     * @param bool   $skipPermissions
+     *
+     * @return array<string, bool>
+     */
+    private function getPermissions(string $permissionsBase, bool $skipPermissions): array
+    {
+        return $skipPermissions
+            ? array_fill_keys(['viewown', 'viewother', 'editown', 'editother'], true)
+            : [
+                'viewown'   => $this->security->isGranted("$permissionsBase:viewown"),
+                'viewother' => $this->security->isGranted("$permissionsBase:viewother"),
+                'editown'   => $this->security->isGranted("$permissionsBase:editown"),
+                'editother' => $this->security->isGranted("$permissionsBase:editother"),
+            ];
```

Comment:
Not all entities have all these permissions. Perhaps we could leave this up to the model to tell us if it has permissions or not? Why do we even need the edit permissions in the search? Users are just viewing, right?

Reply:
> Why do we even need the edit permissions in the search? Users are just viewing, right?

A few search items will help you navigate the entity's edit page.

Reply:
It's OK. We'll show the entities in the search (view) but when accessing the edit page the user should get the access error message.